app/src/org/commcare/mediadownload/MissingMediaDownloadWorker.kt

```diff
@@ -3,21 +3,16 @@ package org.commcare.mediadownload
 import android.content.Context
 import androidx.work.CoroutineWorker
 import androidx.work.WorkerParameters
+import org.commcare.engine.resource.AppInstallStatus
 import org.commcare.resources.model.InstallCancelled
 
 class MissingMediaDownloadWorker(appContext: Context, workerParams: WorkerParameters)
     : CoroutineWorker(appContext, workerParams), InstallCancelled {
 
     override suspend fun doWork(): Result {
         MissingMediaDownloadHelper.installCancelled = this
-        kotlin.runCatching {
-            MissingMediaDownloadHelper.downloadAllLazyMedia()
-        }.onSuccess {
-            return Result.success()
-        }.onFailure {
-            return Result.failure()
-        }
-        return Result.failure()
+        val result = MissingMediaDownloadHelper.downloadAllLazyMedia()
+        return if (result == AppInstallStatus.Installed) Result.success() else Result.retry()
```

Comment:
Don't we only want to retry automatically on specific errors?

In the [handling list above](https://github.com/dimagi/commcare-android/pull/2243/commits/a5fb86158cb77cb120d00ae8a1e34026b9987108#diff-15b34a5a58efc4e31c602ee832bd632eR103), 3 of the 4 errors wouldn't be good for us to run a retry on